pkg/agent/monitoraddnodes.go

```diff
@@ -180,36 +185,65 @@ func (mon *addNodeMonitor) nodeHasJoinedClusterAndIsReady() (bool, bool, error)
 }
 
 func (mon *addNodeMonitor) logCSRsPendingApproval(signerName string) {
-	// TODO? The CSRs have no IP address to identify for which
-	// node it is for, so it is possible to log CSRs pending for
-	// other nodes.
+	csrsPendingApproval := mon.getCSRsPendingApproval(signerName)
+
+	for _, csr := range csrsPendingApproval {
+		mon.logStatus(fmt.Sprintf("CSR %s with signerName %s and username %s is Pending and awaiting approval",
+			csr.Name, csr.Spec.SignerName, csr.Spec.Username))
+	}
+}
+
+func (mon *addNodeMonitor) clusterHasFirstCSRPending() bool {
+	return len(mon.getCSRsPendingApproval(firstCSRSignerName)) > 0
+}
+
+func (mon *addNodeMonitor) clusterHasSecondCSRPending() bool {
+	return len(mon.getCSRsPendingApproval(secondCSRSignerName)) > 0
+}
+
+func (mon *addNodeMonitor) getCSRsPendingApproval(signerName string) []certificatesv1.CertificateSigningRequest {
+	hostnames, err := net.LookupAddr(mon.nodeIPAddress)
+	if err != nil {
+		logrus.Debugf("error looking up hostnames from IP address: %v", err)
```

Comment:
Logging this condition as an `error` may sound a little bit confusing for the user. Since it's a condition that does not prevent the monitor workflow to proceed, I think it could be logged with a message indicating that the hostname cannot be resolved, and thus the CSR checks are skipped.